### DISCUSSION TITLE
Data migration from v2 to v3

### DISCUSSION THREAD

**Discussion by justinaslelys** [Apicurio/apicurio-registry#4821] [General]
Hello,
In the issue #4087 it is mentioned, that automated data migration from 2.x to 3.0 will not be supported.
I understand that schema is not compatible. Then it should be possible to fetch artifacts using API from old instance and re-create using API to new instance of Apicurio. 
The question is if you plan to provide any guidance how to do data migration?

**Comment by EricWittmann** [comment]
Thanks for this question!  And apologies to you and everyone else in the Discussion forum section for our recent silence.  So many communication channels to monitor, and this one got away from me.

Yes, data migration will **not** be automatic between v2 and v3.  However, we **will** have a data migration path that should be convenient.  It goes like this:

1. Export all data from your v2 instance using the `/apis/registry/v2/admin/export` endpoint (this will result in a v2 export ZIP file)
2. Start up your v3 instance
3. Import the ZIP from step (1) into the v3 instance using `/apis/registry/v3/admin/import`

In other words, v3 can import v2 export ZIP files.

Done and done. :)

**Comment by EricWittmann** [comment]
(oops - I assumed this was a Question not a Discussion, which is why I closed it - please re-open if needed!)
